src/nbl/ext/LumaMeter/CLumaMeter.cpp

```diff
@@ -18,37 +17,37 @@ core::SRange<const asset::SPushConstantRange> CLumaMeter::getDefaultPushConstant
 {
 	static const asset::SPushConstantRange range =
 	{
-		ISpecializedShader::ESS_COMPUTE,
+		IShader::ESS_COMPUTE,
 		0u,
 		sizeof(uint32_t)
 	};
 	return {&range,&range+1};
 }
 
-core::SRange<const video::IGPUDescriptorSetLayout::SBinding> CLumaMeter::getDefaultBindings(video::IVideoDriver* driver)
+core::SRange<const video::IGPUDescriptorSetLayout::SBinding> CLumaMeter::getDefaultBindings(video::ILogicalDevice* device)
 {
 	static core::smart_refctd_ptr<IGPUSampler> sampler;
```

Comment:
I know its my sin and my code, the sampler cannot be a `static` the method cannot be static, this all needs to be made in the constructor and return a "cached" version

statics are basically globals, we cannot have them!

Reply:
@nipunG314 thanks to BDA and accessor pattern we don't need to care about bindings or descriptor sets